CanonicalLayouts/list-detail-compose/app/src/main/java/com/example/listdetailcompose/ui/ListDetailSample.kt

```diff
@@ -222,13 +297,43 @@ private fun ListContent(
 private fun DetailContent(
     definedWord: DefinedWord?,
     modifier: Modifier = Modifier,
+    isCompact: Boolean,
+    isDetailVisible: Boolean,
+    sharedTransitionScope: SharedTransitionScope,
+    animatedVisibilityScope: AnimatedVisibilityScope
 ) {
     Column(
         modifier = modifier
             .verticalScroll(rememberScrollState())
             .padding(vertical = 16.dp)
     ) {
         if (definedWord != null) {
+
+            if (isCompact && isDetailVisible) {
+                with(sharedTransitionScope) {
+                    val state = rememberSharedContentState(key = definedWord.word)
+
+                    println("detail-" + definedWord.word + "-" + state.isMatchFound)
+
+                    Image(
+                        painter = painterResource(id = definedWord.icon),
+                        contentDescription = definedWord.word,
+                        modifier = Modifier
+                            .padding(horizontal = 8.dp)
+                            .sharedElement(
```

Comment:
The `.sharedElement` `Modifier` is the only difference between the two places where we output `Modifier`, is it possible to extract that specifically into an `if`/`else` so that we only have a single place where we output `Image`?

If we changed the padding applied around the image in the future, it'd have to be updated in both places, which is a bit of duplication.